### PR TITLE
feat(notion): support notion database

### PR DESCRIPTION
**Pull request by curly210102**
## 该 PR 相关 Issue / Involved Issue

Close #

## 路由地址示例 / Example for the Proposed Route(s)

```routes
/notion/database/a7cc133b68454f138011f1530a13531e
```

## 新 RSS 路由检查表 / New RSS Route Checklist
  
- [x] 新的路由 New Route
  - [x] 跟随 [v2 路由规范](https://docs.rsshub.app/joinus/script-standard.html) Follows [v2 Script Standard](https://docs.rsshub.app/en/joinus/script-standard.html)
- [x] 文档说明 Documentation
  - [x] 中文文档 CN
  - [x] 英文文档 EN
- [x] 全文获取 fulltext
  - [x] 使用缓存 Use Cache
- [ ] 反爬/频率限制 anti-bot or rate limit?
  - [ ] 如果有, 是否有对应的措施? If yes, do your code reflect this sign?
- [x] [日期和时间](https://docs.rsshub.app/joinus/pub-date.html) [date and time](https://docs.rsshub.app/en/joinus/pub-date.html)
  - [x] 可以解析 Parsed
  - [ ] 时区调整 Correct TimeZone
- [x] 添加了新的包 New package added
- [ ] `Puppeteer`

## 说明 / Note
使用说明：需要设置 NOTION_TOKEN 来集成 notion 读取权限
应用场景：Notion作为稍后阅读的存储载体，通过剪藏工具将页面添加至 Notion，读取 Notion 中保存页面列表及页面内容生成一个 feed
添加的新包：
1. @notionhq/client: notion 官方提供的 API SDK
2. notion-to-md：notion 的页面由 blocks 组成，各种类型的 block 解析比较复杂，借助这个包将 blocks 解析并组成 markdown 文本，最后借助 markdown-it 转换为 html

### CODE REVIEW

File: docs/en/install/README.md [hunk 1]
```diff
@@ -832,3 +836,6 @@ See docs of the specified route and `lib/config.js` for detailed information.
 -   ZodGame:
 
     -   `ZODGAME_COOKIE`: Cookie of ZodGame User
+
+-   Notion
+    -   `NOTION_TOKEN`: Notion Internal Integration Token, Refer to [Notion Official Set Up Flow](https://developers.notion.com/docs/authorization#internal-integration-auth-flow-set-up) to create Token
```
Comment (TonyRL) [inline docs/en/install/README.md:841]: Please insert new entries in alphabetical order.

File: docs/en/other.md [hunk 2]
```diff
@@ -356,3 +357,33 @@ Refer to [the list of supported currencies](https://wise.com/tools/exchange-rate
...
+
+::: tip Recommendation
+
+It is recommended to use with clipping tools such as Notion Web Clipper.
+
+:::
+
+### Database
+
+<RouteEn author="curly210102" example="/notion/database/a7cc133b68454f138011f1530a13531e" path="/notion/database/:databaseId" :paramsDesc="['Database ID']" selfhost="1" radar="1" />
```
Comment (TonyRL) [inline docs/en/other.md:377]:
```suggestion
<RouteEn author="curly210102" example="/notion/database/a7cc133b68454f138011f1530a13531e" path="/notion/database/:databaseId" :paramsDesc="['Database ID']" selfhost="1" radar="1">
```

File: docs/en/other.md [hunk 3]
```diff
@@ -356,3 +357,33 @@ Refer to [the list of supported currencies](https://wise.com/tools/exchange-rate
...
+
+For example, if you have set up three properties in your database - "Publish Time", "Author", and "Original Article Link" - then execute the following JavaScript code to get the result for the properties parameter.
+
+```js
+encodeURIComponent(JSON.stringify({"pubTime": "Publish Time", "author": "Author", "link": "Original Article Link"}))
+```
+
+There is an optional query parameter called `query=` that can be used to customize the search rules for your database, such as custom sorting and filtering rules.
+
+please refer to the [Notion API documentation](https://developers.notion.com/reference/post-database-query) and execute `encodeURIComponent(JSON.stringify(custom rules))` to provide the query parameter.
```
Comment (TonyRL) [inline docs/en/other.md:389]:
```suggestion
please refer to the [Notion API documentation](https://developers.notion.com/reference/post-database-query) and execute `encodeURIComponent(JSON.stringify(custom rules))` to provide the query parameter.

</RouteEn>
```

File: docs/install/README.md [hunk 4]
```diff
@@ -971,3 +975,6 @@ RSSHub 支持使用访问密钥 / 码，白名单和黑名单三种方式进行
 
     -   `ZHIHU_COOKIES`: 知乎登录后的 cookie 值.
         1.  可以在知乎网页版的一些请求的请求头中找到，如 `GET /moments` 请求头中的 `cookie` 值.
+
+-   Notion
+    -   `NOTION_TOKEN`: Notion 内部集成 Token，请按照[Notion 官方指引](https://developers.notion.com/docs/authorization#internal-integration-auth-flow-set-up)申请 Token
```
Comment (TonyRL) [inline docs/install/README.md:980]: Please insert new entries in alphabetical order.

File: docs/other.md [hunk 5]
```diff
@@ -1096,3 +1096,33 @@ type 为 all 时，category 参数不支持 cost 和 free
...
+
+::: tip 建议
+
+建议与 Notion Web Clipper 等剪藏工具搭配使用
+
+:::
+
+### Database
+
+<Route author="curly210102" example="/notion/database/a7cc133b68454f138011f1530a13531e" path="/notion/database/:databaseId" :paramsDesc="['Database ID']" selfhost="1" radar="1" />
```
Comment (TonyRL) [inline docs/other.md:1116]:
```suggestion
<Route author="curly210102" example="/notion/database/a7cc133b68454f138011f1530a13531e" path="/notion/database/:databaseId" :paramsDesc="['Database ID']" selfhost="1" radar="1">
```

File: docs/other.md [hunk 6]
```diff
@@ -1096,3 +1096,33 @@ type 为 all 时，category 参数不支持 cost 和 free
...
+
+例如，在 Database 中设置了 "作者"、"文章原链接"、"发布时间"" 三个 property，将下述 JS 代码的执行结果给 `properties` 参数。
+
+```js
+encodeURIComponent(JSON.stringify({"author": "作者", "link": "文章原链接", "pubTime": "发布时间"}))
+```
+
+可选 query 参数 `query=`，用来自定义 Database 的查询规则，例如自定义排序和筛选规则。
+
+书写结构请查阅 [Notion 文档](https://developers.notion.com/reference/post-database-query)，将 `encodeURIComponent(JSON.stringify(自定义规则))` 的执行给 `query` 参数。
```
Comment (TonyRL) [inline docs/other.md:1128]:
```suggestion
书写结构请查阅 [Notion 文档](https://developers.notion.com/reference/post-database-query)，将 `encodeURIComponent(JSON.stringify(自定义规则))` 的执行给 `query` 参数。

</Route>
```

File: lib/config.js [hunk 7]
```diff
@@ -305,6 +305,9 @@ const calculateValue = () => {
         zodgame: {
             cookie: envs.ZODGAME_COOKIE,
         },
+        notion: {
+            key: envs.NOTION_TOKEN,
+        },
```
Comment (TonyRL) [inline lib/config.js:310]: Please insert new entries in alphabetical order.

File: lib/v2/notion/database.js [hunk 8]
```diff
@@ -0,0 +1,160 @@
...
+const { parseDate } = require('@/utils/parse-date');
+const got = require('@/utils/got');
+const { NotionToMarkdown } = require("notion-to-md");
+const cheerio = require('cheerio');
+const md = require('markdown-it')({
+    html: true,
+    linkify: true,
+});
+
+module.exports = async (ctx) => {
```
Comment (TonyRL) [inline lib/v2/notion/database.js:13]: https://github.com/DIYgod/RSSHub/blob/9bdba060c7962d734745c8d6b9264da9a532b51d/lib/routes/disqus/posts.js#L5-L7
Similarly, you can throw an error if the token is not set

File: package.json [hunk 9]
```diff
@@ -76,6 +76,7 @@
   },
   "dependencies": {
     "@koa/router": "12.0.0",
+    "@notionhq/client": "^2.2.5",
```
Comment (TonyRL) [inline package.json:79]:
```suggestion
    "@notionhq/client": "2.2.5",
```

File: package.json [hunk 10]
```diff
@@ -113,6 +114,7 @@
     "mailparser": "3.6.4",
     "markdown-it": "13.0.1",
     "module-alias": "2.2.2",
+    "notion-to-md": "^3.0.1",
```
Comment (TonyRL) [inline package.json:117]:
```suggestion
    "notion-to-md": "3.0.1",
```
The exact semver match is preferred. `pnpm-locked.yaml` needs to be updated as well.

File: lib/v2/notion/database.js (hunk 8 above)
Comment (curly210102) [reply lib/v2/notion/database.js:13]: Done, I've added notion key check

**Comment by curly210102** [comment]
> Successfully [generated](https://github.com/DIYgod/RSSHub/actions/runs/5045546702) as following:
> 
> http://localhost:1200/notion/database/a7cc133b68454f138011f1530a13531e - Failed ❌

The test NOTION_TOKEN=secret_on0n0W5QUDgY7ZaTUGFKJ4W8XcahrfMgJi8Jfwefnk

**Pull request merged**